### PR TITLE
docs: add changelog for 4.2.0

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

### CODE REVIEW

**Review by lengau — APPROVED** [review]

File: docs/reference/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
 
+4.2.0 (2024-Sep-12)
+-------------------
+
+Application
+===========
+
+- Add a configuration service to unify handling of command line arguments,
+  environment variables, snap configuration, etc.
+- Properly support ``--shell``, ``--shell-after`` and ``--debug`` on the
```
Comment (lengau) [inline docs/reference/changelog.rst:13]: This might be better under a "Lifecycle commands" subheader or something?

Comment (tigarmo) [reply docs/reference/changelog.rst:13]: how about just a "Commands" subheader?

**Review by mr-cal — APPROVED** [review]

File: docs/reference/changelog.rst [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
 *********
 
+4.2.0 (2024-Sep-12)
+-------------------
+
+Application
+===========
+
+- Add a configuration service to unify handling of command line arguments,
+  environment variables, snap configuration, etc.
```
Comment (medubelko) [inline docs/reference/changelog.rst:12]:
```suggestion
  environment variables, snap configurations, and so on.
```

File: docs/reference/changelog.rst [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+- Properly support ``--shell``, ``--shell-after`` and ``--debug`` on the
+  ``pack`` command.
+- Use the standard library to get proxies set on the host.
```
Comment (medubelko) [inline docs/reference/changelog.rst:15]:
```suggestion
- Use the standard library to retrieve the host's proxies.
```

**Review by medubelko — CHANGES_REQUESTED** [review]

**Review by medubelko — APPROVED** [review]

**Pull request merged**